doc: feature management proposal (#41420)
nodejs/node: doc/guides/feature-request-management.md

# Feature request management

Feature requests are a valuable source of input to the project.
They help our maintainers understand what additions will be of
value to users of the Node.js runtime.

At the same time, the project is volunteer run and does not

> have the ability to `direct` resource at specific work. The

Reviewer: ```suggestion
have the ability to direct resources toward specific work. The
```

features which are implemented are those for which volunteers
are individually motivated to work on. The best way to ensure
a feature gets implemented is to create a PR to add it.

> The project stives to support people who do that.

Reviewer 2: ```suggestion
The project strives to support people who do that.
```

An open feature request does not provide any indication that work
on that feature will take place and after some period of time
may be detrimental as it may result in an expectation that will

> never be fullfilled.

Reviewer: ```suggestion
never be fulfilled.
```

This process tries to balance retaining the valuable input
we get through feature requests and the overhead of

> maintaining open features requests that will never get

Reviewer: ```suggestion
maintaining open feature requests that will never get
```

implemented.

## Creating feature requests

A feature request can be created by adding the `feature request`

> lable to an issue. This may be done when the issue is opened or

Reviewer: ```suggestion
label to an issue. This may be done when the issue is opened or
```

> A feature request can be created by adding the `feature request`
> lable to an issue. This may be done when the issue is opened or
> at a later point (often when investigation of a bug report
> results in it being considered a feature request as opposed to a
> bug).

Reviewer 2: ```suggestion
A feature request can be created by adding the `feature request`
label to an issue. This may be done automatically when the issue 
is opened or at a later point (often when investigation of a bug
report results in it being considered a feature request as opposed
to a bug).
```

> The current list of feature requests can be found through this
> [query](https://github.com/nodejs/node/issues?q=is%3Aissue+is%3Aopen+label%3A%22feature+request%22).

Reviewer 2: ```suggestion
The current list of feature requests can be found through the
[is:issue is:open label:"feature request"](https://github.com/nodejs/node/issues?q=is%3Aissue+is%3Aopen+label%3A%22feature+request%22) query.
```

## Triage of feature requests

There is no set process for triaging/handling feature requests.
Individual collaborators review issues marked as `feature request`

> along with other issues and may or may not decided to

Reviewer: ```suggestion
along with other issues and may or may not decide to
```

work on an implementation or advocate on their behalf.

If a collaborator believes a feature request must be implemented
they can add the `never-stale` label to the issue and it will
be excluded from the automated feature request handling
as outlined below.

> ## Expressing support for a feature request.

Reviewer 3: ```suggestion
## Expressing support for a feature request
```

If you come across a feature request and want to add your

> support for that feature please express your suport

Reviewer: ```suggestion
support for that feature please express your support
```

> though through the thumbs up emoji. At some point in the

Reviewer 2: Do you mean by commenting or by reacting? I assume by reacting, but probably good to clarify.

Author: ```suggestion
though through the thumbs up emoji as a reaction. At some point in the
```

@bnb is this what you had in mind?

> though through the thumbs up emoji. At some point in the

Reviewer: ```suggestion
with the thumbs up emoji. At some point in the
```

Also I agree with the need for clarification here about commenting/reacting.

Author: Fixed in update above.

future we may use this as additional input in the automated
handling of feature requests.

## Automated feature request handling

> Our experience is that most feature requests that are

Reviewer 4: Why are these line lengths so short? :-P

Author: I prefer that to having to check each one for being too long :)

likely to be addressed, will be addressed within the first
6 months after they are submitted.

Once there has been no activity on a feature request for
5 months, the following comment will be added
to the issue:

> ````text

Reviewer 2: ```suggestion
```text
```

> ````text

Reviewer 5: `````suggestion
````markdown
`````

> ```text

Reviewer 5: ````suggestion
```markdown
````

There has been no activity on this feature request for
5 months and it is unlikely to be implemented.

> It will be closed after 6 months unless there is renewed

Reviewer: This time frame should be clarified. It seems a bit ambiguous as to whether it's 6 months from the time the comment is posted, since the last (non-automated) comment, or something else.

Author: @mscdex done.

> It will be closed after 6 months unless there is renewed

Author: ```suggestion
It will be closed after 6 months after the non-automated comment unless there is renewed
```

discussion or collaborators advocating that it be kept
open. For more information on how the project manages

> feature requests please consult
> [feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).

Reviewer: ```suggestion
feature requests please consult the
[feature request management document](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
```

> 

Reviewer 2: need to include ``` here

> ````text
> There has been no activity on this feature request for
> 5 months and it is unlikely to be implemented.
> It will be closed after 6 months unless there is renewed
> discussion or collaborators advocating that it be kept
> open. For more information on how the project manages
> feature requests please consult
> [feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
> 

Reviewer 3: ```suggestion
\`\`\`text
There has been no activity on this feature request for
5 months and it is unlikely to be implemented.
It will be closed after 6 months unless there is renewed
discussion or collaborators advocating that it be kept
open. For more information on how the project manages
feature requests please consult
[feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
\`\`\`
```

Omit the backslashes before adding this suggestion as a commit, they are there to prevent the suggestion from being rendered incorrectly.

Reviewer 5: I think Micheal meant to use 4 backticks here. In markdown, you can use more than 3 backticks to delimite the `<pre>` section, it's especially useful when rendering markdown inside markdown. That also work for suggestions in GitHub btw.

`````markdown
````suggestion
```text
Lorem ipsum
```
````
`````

If there is no additional activity/discussion on the
feature request in the next month, the following
comment is added to the issue and the issue will be
closed:

> ```text

Reviewer 5: ````suggestion
```markdown
````

and it is being closed.  We value your input
and if you feel closing this issue was not the

> right thing to do please re-open it.

Reviewer: ```suggestion
right thing to do, please re-open it.
```

open.

For more information on how the project manages

> feature requests please consult
> [feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).

Reviewer: ```suggestion
feature requests please consult the
[feature request management document](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
```

> ````

Reviewer 2: code block formatting is off here. I think you're missing a set of triple backticks and have one too many backticks in the first and last set?

Author: @bnb fixed

> ```text
> There has been no activity on this feature request
> and it is being closed.  We value your input
> and if you feel closing this issue was not the
> right thing to do please re-open it.
> open.
> 
> For more information on how the project manages
> feature requests please consult
> [feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
> ````

Reviewer 3: ```suggestion
\`\`\`text
There has been no activity on this feature request
and it is being closed.  We value your input
and if you feel closing this issue was not the
right thing to do please re-open it.
open.

For more information on how the project manages
feature requests please consult
[feature request management](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
\`\`\`
```
Omit the backslashes before adding this suggestion as a commit, they are there to prevent the suggestion from being rendered incorrectly.

Reviewer 5: ````suggestion
```markdown
There has been no activity on this feature request
and it is being closed. We value your input
and if you feel closing this issue was not the
right thing to do, please re-open it.

For more information on how the project manages
feature requests please consult the
[feature request management document](https://github.com/nodejs/node/blob/master/doc/guides/feature-request-management.md).
```
````
